lib/improver/psychrometric_calculations/psychrometric_calculations.py

```diff
@@ -148,25 +146,24 @@ def calculate_d_enthalpy_dt(mixing_ratio, specific_heat,
             last updated 2014-12-05.
 
         Args:
-            mixing_ratio (iris.cube.Cube):
+            mixing_ratio (numpy.ndarray):
                 Cube of mixing ratios.
-            specific_heat (iris.cube.Cube):
+            specific_heat (numpy.ndarray):
                 Cube of specific heat capacities of moist air (J kg-1 K-1).
-            latent_heat (iris.cube.Cube):
+            latent_heat (numpy.ndarray):
                 Cube of latent heats of condensation of water vapour
                 (J kg-1).
-            temperature_input (iris.cube.Cube):
+            temperature (numpy.ndarray):
                 A cube of temperatures (K, or converted).
 
         Returns:
-            iris.cube.Cube:
+            numpy.ndarray:
                 A cube of the enthalpy gradient with respect to temperature.
         """
-        temperature = temperature_input.copy()
-        temperature.convert_units('K')
-        numerator = (mixing_ratio * latent_heat ** 2)
-        denominator = consts.U_R_WATER_VAPOUR * temperature ** 2
-        return numerator/denominator + specific_heat
+        numerator = mixing_ratio * latent_heat ** 2
+        denominator = consts.U_R_WATER_VAPOUR.points * temperature ** 2
+        dived = numerator/denominator
```

Comment:
Why the intermediate calculation?

Reply:
I don't know. Fixed now.
there is a minimal error when removing `.points` so that needs a proper look at or fix.